### PR TITLE
Removing async wait handle

### PR DESCRIPTION
**Pull request by jasonsandlin**

### CODE REVIEW

File: Include/httpClient/async.h [hunk 1]
```diff
@@ -64,6 +57,11 @@ typedef struct AsyncBlock
     /// </summary>
     AsyncCompletionRoutine* callback;
 
+    /// <summary>
+    /// Internal use only
+    /// </summary>
+    HANDLE internalHandle;
```
Comment (brianpepin) [inline Include/httpClient/async.h]: Not necessary.  We create one internally and can just use that.

File: Source/Task/AsyncLib.cpp [hunk 2]
```diff
@@ -438,10 +438,10 @@ static void CALLBACK CompletionCallbackForAsyncBlock(
     {
         asyncBlock->callback(asyncBlock);
     }
-    if (asyncBlock->waitEvent)
+    if (asyncBlock->internalHandle)
```
Comment (brianpepin) [inline Source/Task/AsyncLib.cpp]: Woah. I think this is all wrong (existing code).  Note that when we schedule the callback we pass in a state object, not an async block. This should attach to a state pointer, invoke with the state pointer's async block and then set the event in the state pointer.

File: Include/httpClient/async.h (hunk 1 above)
Comment (lubeltra) [reply Include/httpClient/async.h]: Agreed

---
In reply to: [193618417](https://github.com/Microsoft/libHttpClient/pull/186#discussion_r193618417) [](ancestors = 193618417)

**Comment by lubeltra** [comment]
>     }

This code could just be removed and state can create the event internally

---
Refers to: Source/Task/AsyncLib.cpp:283 in 9d6b317. [](commit_id = 9d6b317ae2ad9ad2f4138a70293bf40c95422f10, deletion_comment = False)

File: Source/Task/AsyncLib.cpp (hunk 2 above)
Comment (lubeltra) [reply Source/Task/AsyncLib.cpp]:
```c++
            hr = SubmitAsyncCallback(
                asyncBlock->queue,
                AsyncQueueCallbackType_Completion,
                asyncBlock,
                CompletionCallbackForAsyncBlock);
```
context is the AsyncBlock

---
In reply to: [193618839](https://github.com/Microsoft/libHttpClient/pull/186#discussion_r193618839) [](ancestors = 193618839)

**Pull request merged**